components/merkledb/src/macros.rs

```diff
@@ -58,3 +60,28 @@ macro_rules! impl_object_hash_for_binary_value {
         )*
     };
 }
+
+// Think about bincode instead of protobuf. [ECR-3222]
+/// Implements `BinaryKey` trait for any type that implements `BinaryValue`.
+#[macro_export]
+macro_rules! impl_binary_key_for_binary_value {
+    ($type:ty) => {
+        impl exonum_merkledb::BinaryKey for $type {
+            fn size(&self) -> usize {
+                exonum_merkledb::BinaryValue::to_bytes(self).len()
+            }
+
+            fn write(&self, buffer: &mut [u8]) -> usize {
+                let bytes = exonum_merkledb::BinaryValue::to_bytes(self);
+                buffer.copy_from_slice(&bytes);
+                bytes.len()
+            }
+
```

Comment:
What if a `copy_from_slice` replace with a `swap_with_slice` method. I think it can bring more performance.
```rust
     fn write(&self, buffer: &mut [u8]) -> usize {
                let mut bytes = exonum_merkledb::BinaryValue::to_bytes(self);
                buffer.swap_from_slice(&mut bytes);
                bytes.len()
            }
```